drivers/clock_control/clock_control_mcux_sim.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2017, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <errno.h>
+#include <soc.h>
+#include <clock_control.h>
+#include <fsl_clock.h>
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_CLOCK_CONTROL_LEVEL
+#include <logging/sys_log.h>
+
+static inline int mcux_sim_on(struct device *dev,
```

Comment:
I think this should be just static function no need of inline as it will be called lie clk->on()/off.
making it inline doesn't look logical to me.

Reply:
Agree, will fix. This was a copy/paste mistake, you can see it in the other clock_control drivers.